addons/account/models/account_move.py

```diff
@@ -1713,18 +1713,50 @@ def _check_duplicate_supplier_reference(self):
             ))
 
     def _check_balanced(self):
-        ''' Assert the move is fully balanced debit = credit.
-        An error is raised if it's not the case.
-        '''
+        """ Process request to check that moves are balanced: either do it immediately
+            or postpone till the end of transaction for performance sake"""
+
         moves = self.filtered(lambda move: move.line_ids)
         if not moves:
             return
 
+        # TODO in master: flushing should be done only if we need to check balance
+        # immediately, but apparently some other accounting code depends on
+        # this flushing; so always do the flushing to don't break standard or custom modules
+
         # /!\ As this method is called in create / write, we can't make the assumption the computed stored fields
         # are already done. Then, this query MUST NOT depend of computed stored fields (e.g. balance).
         # It happens as the ORM makes the create with the 'no_recompute' statement.
         self.env['account.move.line'].flush(self.env['account.move.line']._fields)
         self.env['account.move'].flush(['journal_id'])
+
+        if not self.env.context.get("account.check_balanced_immediatly"):
+            return moves._postpone_check_balanced()
```

Comment:
When is this used?

Reply:
it's not used for now

Reply:
?

Reply:
It may be needed, e.g. if you have a code that catches UserError. Some custom modules, for example

Reply:
At least change the key then, the attribute is annoying do do a `with_context(key=attribute)` because of the dot

Reply:
Indeed  :+1: . I was wondering why nobody uses dots in context